specification/apimanagement/resource-manager/Microsoft.ApiManagement/2017-03-01/apimapis.json

```diff
@@ -1399,15 +1660,174 @@
         },
         "isCurrent": {
           "type": "boolean",
-          "description": "Indicates if API revision is current api revision."
+          "description": "Indicates if API revision is current api revision.",
+          "readOnly": true
         },
         "isOnline": {
           "type": "boolean",
-          "description": "Indicates if API revision is accessible via the gateway."
+          "description": "Indicates if API revision is accessible via the gateway.",
+          "readOnly": true
         }
       },
       "description": "API base contract details."
     },
+    "ApiRevisionCollection": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ApiRevisionContract"
+          },
+          "description": "Page values."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "Next page link if any."
+        }
+      },
+      "description": "Paged Api Revision list representation."
+    },
+    "ApiRevisionContract": {
+      "properties": {
+        "apiId": {
+          "type":"string",
+          "readOnly": true,
+          "description": "Identifier of the API Revision."
+        },
+        "apiRevision": {
+          "type":"string",
+          "readOnly": true,
+          "description": "Revision number of API."
+        },
+        "createdDateTime": {
+          "type":"string",
+          "readOnly": true,
+          "format": "date-time",
+          "description": "The time the API Revision was created."
+        },
+        "updatedDateTime": {
+          "type":"string",
+          "readOnly": true,
+          "format": "date-time",
+          "description": "The time the API Revision were updated."
+        },
+        "description": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Description of the API Revision."
```

Comment:
>Description [](start = 26, length = 11)

how long can the description be

Reply:
Not sure. I'll copy it from the ApiRevisionDescription in ApiContract.